docs/source/architecture/sub_stakes.rst

```diff
@@ -0,0 +1,276 @@
+
+A sub-stake remains active until it becomes unlocked, and a staker gets the reward for the last period by calling ``mint()`` or ``confirmActivity()`` once the last period is surpassed. Each staker can have no more than 30 active sub-stakes which are stored in an array. All sub-stake changes initially reuse slots of inactive sub-stakes for storage in the array, and if there are none, will instead use empty slots. Therefore, attempting to retrieve data about previous inactive sub-stakes is not guaranteed to be successful since the data could have been overwritten.
+
+
+
+Operations that modify the sub-stake array
+------------------------------------------
+
+Deposit and locking
+^^^^^^^^^^^^^^^^^^^
+*Methods* : ``deposit(uint256,uint16)``,  ``deposit(address,uint256,uint16)``,  ``lock(uint256,uint16)``
+
+To become a staker, NU tokens must be transferred to the ``StakingEscrow`` contract and locked using one of the ``deposit()`` methods. If the staker already has unlocked tokens within the account in the contract, then the stake can be locked using the ``lock()`` method. If successful, a new element will be created in the array of sub-stakes with the next period as the starting date, and the duration equal to the input parameter.
+
+
+**Example:**
+
+    A staker deposits 900 tokens:
+        * 1st sub-stake = 900 tokens starting from the next period and locked for 5 periods
+
+    .. aafig::
+        :proportional:
+        :textual:
+
+            stake
+            ^
+            |
+         900|   +-------------------+
+            |   |        1st        |   period
+            +---+---+---+---+---+---+---->
+            | 0 | 1 | 2 | 3 | 4 | 5 | 6
+
+
+
+Lock prolongation
+^^^^^^^^^^^^^^^^^
+*Methods* : ``prolongStake(uint256,uint16)``
+
+In order to increase the staking reward, as well as the possibility of obtaining policies with a longer timeframe, stakers can increase the duration of their locked sub-stake using the ``prolongStake()`` method. The number of sub-stakes does not change, but the locked duration for the specified sub-stake will be increased.
+
+**Example:**
+
+    A staker prolongs sub-stake for 2 additional periods up to 7:
+		- Before: 
+			* 1st sub-stake = 900 tokens for 5 periods
+		- After: 
+			* 1st sub-stake = 900 tokens for 7 periods
+
+    .. aafig::
+        :proportional:
+        :textual:
+
+                         Before             
+
+            stake
+            ^
+            |
+         900+-----------------------+
+            |          1st          |   period
+            +---+---+---+---+---+---+---->
+            | 0 | 1 | 2 | 3 | 4 | 5 | 6      
+			
```

Comment:
@vzotova This is where I get confused - (and apologies, if I'm just totally missing it) - but shouldn't the first sub-stake for the `before` scenario stop at the end of period 4, and not at the end of period 5? i.e. the 1st sub-stake includes the current period and is for periods 0, 1, 2, 3, 4 (total = 5 periods). 

This stake starts from the current period (0), no? or is it 5 periods starting the next period, like in the `Deposit and locking` scenario above, in which case it should start from period 1?

Reply:
current period does not count, it's sort of historical data, for example lock for 0 periods looks like
```
^
|
+---+
|   |
+---+--->
| 0 | 1
```

Reply:
Perhaps it boils down to a wording issue - we should indicate whether the current period is included or not, eg.

```
* 1st sub-stake = 900 tokens for 5 periods (in addition to the current period)
```

wdyt? Open to alternative wording as well.

@arjunhassard wdyt about the wording?

Reply:
I'm for describing this somewhere in the first section, because current period always does not count when we talk about locking duration

Reply:
👍 - Adding it as an overall statement makes sense.

Reply:
I agree that it'd be nice for some additional clarity on this. You can see from my other comments that I'm quite confused about it 😆

Reply:
I'm a little bit stuck with formulating this, could you help me
What I want to say:
* duration (phrases "locked for ") means how many times need to call `confirmActivity()` so this leads to thought that duration is only future periods (current does not count)
* locking in the current period depends on waht situation was in past - if we step 1 period to the previous period and look into next one (which is current period in the current moment), if next period is locked for some sub-stake that means lock in the current period in the current moment and vice versa

Reply:
So just to ensure that I understand, we define `locking duration` as the number of times `confirmActivity()` needs to be called.

If a stake has a locking duration of 5 periods it means that the stake will be active for periods: 0,1,2,3,4,5 because `confirmActivity()` would be called during periods 0,1,2,3,4 each time confirming activity for the subsequent period (1,2,3,4,5) respectively.

Is this right?

Reply:
almost perfect, 0th period lock is not guaranteed, so lock will be active for periods: 1,2,3,4,5 and 0th period should be specified additionally (which by default is locked and unlocked when said that lock starts in the next period)

Reply:
summing up, the correct explanation:
`If a stake has a locking duration of 5 periods it means that the stake will be active for periods: 1,2,3,4,5 because confirmActivity() would be called during periods 0,1,2,3,4 each time confirming activity for the subsequent period (1,2,3,4,5) respectively.`

Reply:
Right, so unless specified otherwise (example by stating "*starting from the next period *") the 0th period is considered active.

---

@vzotova what about something like the following at the top of the doc (after the sentence that ends "and can be freely withdrawn.").

```
[and can be freely withdrawn.]

Each sub-stake has a beginning and locking duration. For our terminology, we define `locking duration` to be the required number of times that `confirmActivity()` needs to be called so that the stake is registered as active for the subsequent period. For example, if a stake has a *locking duration of 5 periods* it means that the stake will be active for periods: 1,2,3,4,5 because `confirmActivity()` would be called during periods 0,1,2,3,4 each time confirming activity for the subsequent period (1,2,3,4,5) respectively. Unless the starting period is specified, the stake is considered active for the 0th (current) period by default.
```

Obviously you can reword as needed.

Reply:
I was also thinking we could remove the sentence at the top (Overview):

` When a staker confirms activity each day and winding down is enabled, the remaining lock time for relevant sub-stakes is reduced.` 

since wind-down will be explained appropriately later on.

Reply:
Changed `active` to `locked` because we introduce `active` in the next paragraph with other meaning. and also `stake` -> `sub-stake` (here it's very important).
About winding down: sometimes better to say twice :) but if you sure - I can remove